examples/13_rag.py

```diff
@@ -0,0 +1,174 @@
+4. Provides well-structured, informative responses
+
+Note: WorkflowAI does not manage the RAG implementation (yet). You need to provide your own
+search implementation to connect to your knowledge base. This example uses a mock search
+function to demonstrate the pattern.
+"""
+
+import asyncio
+from enum import Enum
+from typing import List, Optional
+
+from pydantic import BaseModel, Field
+
+import workflowai
+from workflowai import Model, Run
+
+
+class Role(str, Enum):
+    """Enum representing possible message roles."""
+    USER = "user"
+    ASSISTANT = "assistant"
+
+
+class SearchResult(BaseModel):
+    """Model representing a search result from the knowledge base."""
+    content: str = Field(
+        description="The content of the search result",
+    )
+    relevance_score: float = Field(
+        description="Score indicating how relevant this result is to the query",
+    )
+
+
+# Simulated knowledge base search tool
+# ruff: noqa: ARG001
+async def search_faq(query: str) -> List[SearchResult]:
+    """
+    Search the knowledge base for relevant information.
+
+    Args:
+        query: The search query to find relevant information
+
+    Returns:
+        A list of search results ordered by relevance
+    """
+    # This is a mock implementation - in a real system this would query your knowledge base
+    # The results below are hardcoded but in a real implementation would be based on the query
+    return [
+        SearchResult(
+            content=(
+                "Our standard return policy allows returns within 30 days of purchase with original "
+                "receipt. Items must be unused and in original packaging. Once received, refunds are "
+                "processed within 5-7 business days."
+            ),
+            relevance_score=0.95,
+        ),
+        SearchResult(
+            content=(
+                "For online purchases, customers can initiate returns through their account dashboard "
+                "or by contacting customer support. Free return shipping labels are provided for "
+                "defective items."
+            ),
+            relevance_score=0.88,
+        ),
+        SearchResult(
+            content=(
+                "Standard shipping takes 3-5 business days within the continental US. Express shipping "
+                "(1-2 business days) is available for an additional fee. Free shipping on orders over $50."
+            ),
+            relevance_score=0.82,
+        ),
+    ]
+
+
+class Message(BaseModel):
+    """Model representing a chat message."""
+    role: Role = Field()
+    content: str = Field(
+        description="The content of the message",
+    )
+
+
+class AssistantMessage(Message):
+    """Model representing a message from the assistant."""
+    role: Role = Role.ASSISTANT
+    content: str = ""
+
+
+class ChatbotOutput(BaseModel):
+    """Output model for the chatbot response."""
+    assistant_message: AssistantMessage = Field(
```

Comment:
```suggestion
    assistant_message: AssistantMessage = Field(default_factory=AssistantMessage,
```

Reply:
It would provide a default value for `assistant_message` which would make it accessible while streaming

See this section of the readme https://github.com/WorkflowAI/workflowai-py?tab=readme-ov-file#required-versus-optional-fields